rmw_opensplice_cpp/src/rmw_publisher.cpp

```diff
@@ -170,6 +170,13 @@ rmw_create_publisher(
 
   publisher->implementation_identifier = opensplice_cpp_identifier;
   publisher->data = publisher_info;
+  publisher->topic_name = reinterpret_cast<const char *>(
+    rmw_allocate(sizeof(char) * strlen(topic_name) + 1));
```

Comment:
For platform where `sizeof(char)` is not `1` the memcpy code below will not work anyway.